stdlib/strings/strings.go

```diff
@@ -382,7 +385,77 @@ func generateUnicodeIsFunction(name string, Fn func(rune) bool) values.Function
 	)
 }
 
+var strlen = values.NewFunction(
+	"strlen",
+	semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{stringArgV: semantic.String},
+		Required:   semantic.LabelSet{stringArgV},
+		Return:     semantic.Int,
+	}),
+	func(args values.Object) (values.Value, error) {
+		v, ok := args.Get(stringArgV)
+		if !ok {
+			return nil, fmt.Errorf("missing argument %q", stringArgV)
+		}
+
+		if v.Type().Nature() == semantic.String {
+			return values.NewInt(int64(utf8.RuneCountInString(v.Str()))), nil
+		}
+
+		return nil, fmt.Errorf("procedure cannot be executed")
+	}, false,
+)
+
+var substring = values.NewFunction(
+	"substring",
+	semantic.NewFunctionPolyType(semantic.FunctionPolySignature{
+		Parameters: map[string]semantic.PolyType{
+			stringArgV: semantic.String,
+			start:      semantic.Int,
+			end:        semantic.Int},
+		Required: semantic.LabelSet{stringArgV, start, end},
+		Return:   semantic.String,
+	}),
+	func(args values.Object) (values.Value, error) {
+		v, vOk := args.Get(stringArgV)
+		a, aOk := args.Get(start)
+		b, bOk := args.Get(end)
+		if !aOk || !bOk || !vOk {
+			return nil, fmt.Errorf("missing argument")
+		}
+
+		vStr := v.Str()
+		aInt := int(a.Int())
+		bInt := int(b.Int())
+		if aInt < 0 || bInt > len(vStr) {
+			return nil, fmt.Errorf("indices out of bounds")
+		}
+
+		if (v.Type().Nature() == semantic.String) && (a.Type().Nature() == semantic.Int) && (b.Type().Nature() == semantic.Int) {
```

Comment:
This type check needs to happen before lines 427-429. Otherwise those methods will panic.